### PR TITLE
Added support for Mock XML request and responses.

### PR DESCRIPTION
**Pull request by gagarwal**
I have added support for Mock XML request and responses. I still need to implement Unit Tests for new implemented API's. I have made sure the existing Unit Tests are passing fine.

Feedback is most welcome.

### CODE REVIEW

File: URLMock/Mock Messages/UMKMockHTTPMessage.h [hunk 1]
```diff
@@ -108,6 +114,14 @@ extern NSString *const kUMKMockHTTPMessageUTF8WWWFormURLEncodedContentTypeHeader
 /*! @methodgroup Geting and setting the body */
 
 /*!
+ @abstract Sets the receiver's body to the specified XML string.
+ @discussion If the receiver does not already have a value for the Content-type header field, sets the value of that
+ header to "application/xml; charset=utf-8".
```
Comment (prachigauriar) [inline URLMock/Mock Messages/UMKMockHTTPMessage.h]: Indent this second line by four spaces.

File: URLMock/Mock Messages/UMKMockHTTPMessage.h [hunk 2]
```diff
@@ -108,6 +114,14 @@ extern NSString *const kUMKMockHTTPMessageUTF8WWWFormURLEncodedContentTypeHeader
...
+ @param xmlString The XML string set as the receiver's body.
+ */
+- (void)setBodyWithXMLString:(NSString *)xmlString;
```
Comment (prachigauriar) [inline URLMock/Mock Messages/UMKMockHTTPMessage.h]: Capitalize XML in XMLString.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 3]
```diff
@@ -31,30 +31,148 @@
 
 @interface UMKMockURLProtocol (UMKHTTPConvenienceMethods)
 
-/*! @methodgroup Mock Requests with JSON Bodies with Error Responders */
+/*! @methodgroup Mock Requests without Bodies and with Error Responders */
+
+/*!
+ @abstract Creates and expects a new mock HTTP GET request that responds with an error.
+ @discussion There is no need to add the returned mock request to the list of expected mock requests, as
+ this method adds it automatically.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: Indent this line by four spaces.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 4]
```diff
@@ -31,30 +31,148 @@
...
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPGetRequestWithURL:(NSURL *)URL responseError:(NSError *)error;
+
+/*! @methodgroup Mock Requests with XML Bodies with Error Responders */
 
 /*!
  @abstract Creates and expects a new mock HTTP request that responds with an error.
  @discussion There is no need to add the returned mock request to the list of expected mock requests, as
-     this method adds it automatically.
+ this method adds it automatically.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: Indent by four spaces.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 5]
```diff
@@ -31,30 +31,148 @@
...
  @param method The HTTP method for the mock HTTP request. May not be nil.
  @param URL The URL for the mock HTTP request. May not be nil.
- @param requestJSON A JSON object to use as the mock HTTP request body.
+ @param requestXML A XML string to use as the mock HTTP request body.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: *An* XML string.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 6]
```diff
@@ -31,30 +31,148 @@
...
 + (UMKMockHTTPRequest *)expectMockHTTPRequestWithMethod:(NSString *)method URL:(NSURL *)URL
-                                            requestJSON:(id)requestJSON responseError:(NSError *)error;
+                                             requestXML:(NSString *)requestXML responseError:(NSError *)error;
 
 /*!
- @abstract Creates and expects a new mock HTTP GET request that responds with an error.
+ @abstract Creates and expects a new mock HTTP PATCH request that responds with an error.
  @discussion There is no need to add the returned mock request to the list of expected mock requests, as
-     this method adds it automatically.
+ this method adds it automatically.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: Indent 4 spaces.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 7]
```diff
@@ -31,30 +31,148 @@
...
+ @param URL The URL for the mock PATCH request. May not be nil.
+ @param requestXML A XML string to use as the mock HTTP PATCH request body.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: *An* XML string.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 8]
```diff
@@ -31,30 +31,148 @@
...
+ @param error The error that the mock responder responds with. May not be nil.
+ @result The newly created mock HTTP PATCH request. Its responder is set to the new mock error responder.
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPPatchRequestWithURL:(NSURL *)URL requestXML:(NSString *)requestXML responseError:(NSError *)error;
+
+/*!
+ @abstract Creates and expects a new mock HTTP POST request that responds with an error.
+ @discussion There is no need to add the returned mock request to the list of expected mock requests, as
+ this method adds it automatically.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: Indent by 4 spaces.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 9]
```diff
@@ -31,30 +31,148 @@
...
+ @param URL The URL for the mock POST request. May not be nil.
+ @param requestXML A XML string to use as the mock HTTP POST request body.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: *An* XML string.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 10]
```diff
@@ -31,30 +31,148 @@
...
+ @param error The error that the mock responder responds with. May not be nil.
+ @result The newly created mock HTTP POST request. Its responder is set to the new mock error responder.
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPPostRequestWithURL:(NSURL *)URL requestXML:(NSString *)requestXML responseError:(NSError *)error;
+
+/*!
+ @abstract Creates and expects a new mock HTTP PUT request that responds with an error.
+ @discussion There is no need to add the returned mock request to the list of expected mock requests, as
+ this method adds it automatically.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: Indent by 4 spaces.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 11]
```diff
@@ -31,30 +31,148 @@
...
+ @param URL The URL for the mock PUT request. May not be nil.
+ @param requestXML A XML string to use as the mock HTTP PUT request body.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: *An* XML string…

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 12]
```diff
@@ -31,30 +31,148 @@
...
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPPutRequestWithURL:(NSURL *)URL requestXML:(NSString *)requestXML responseError:(NSError *)error;
+
+
+/*! @methodgroup Mock Requests and Responders with XML Bodies */
+
+/*!
+ @abstract Creates and expects a new mock HTTP request that responds with a XML string.
+ @discussion There is no need to add the returned mock request to the list of expected mock requests, as
+ this method adds it automatically.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: Always make the second line of a HeaderDoc comment indented by 4 spaces.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 13]
```diff
@@ -31,30 +31,148 @@
...
+ @param method The HTTP method for the mock HTTP request. May not be nil.
+ @param URL The URL for the mock HTTP request. May not be nil.
+ @param requestXML A XML string to use as the mock HTTP request body.
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: Replace all “A XML” with “An XML”.

File: URLMock/Mock Messages/UMKMockHTTPMessage.h (hunk 2 above)
Comment (prachigauriar) [inline URLMock/Mock Messages/UMKMockHTTPMessage.h]: Also, move this below the URLEncoded body methods.

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h [hunk 14]
```diff
@@ -31,30 +31,148 @@
...
+ @discussion There is no need to add the returned mock request to the list of expected mock requests, as
+ this method adds it automatically.
+ @param URL The URL for the mock PUT request. May not be nil.
+ @param requestXML A XML string to use as the mock HTTP PUT request body.
+ @param statusCode The status code for the HTTP response.
+ @param responseXML A XML string to use as the mock HTTP response body.
+ @result The newly created mock HTTP PUT request. Its responder is set to the new mock XML body responder.
+ */
++ (UMKMockHTTPRequest *)expectMockHTTPPutRequestWithURL:(NSURL *)URL requestXML:(NSString *)requestXML
+                                     responseStatusCode:(NSInteger)statusCode responseXML:(NSString *)responseXML;
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.h]: Move all these methods below the JSON ones.

Also, where you‘ve been saying requestXML: and responseXML:, rename those to requestXMLString: and responseXMLString:

File: URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.m [hunk 15]
```diff
@@ -32,6 +32,99 @@
...
+
+
++ (UMKMockHTTPRequest *)expectMockHTTPPutRequestWithURL:(NSURL *)URL requestXML:(NSString *)requestXML
+                                     responseStatusCode:(NSInteger)statusCode responseXML:(NSString *)responseXML
+{
+    return [self expectMockHTTPRequestWithMethod:kUMKMockHTTPRequestPutMethod URL:URL requestXML:requestXML
+                              responseStatusCode:statusCode responseXML:responseXML];
+}
+
+
```
Comment (prachigauriar) [inline URLMock/Mock URL Protocol/UMKMockURLProtocol+UMKHTTPConvenienceMethods.m]: Move al this stuff below the JSON methods.

**Comment by prachigauriar** [comment]
@gagarwal: This looks great. Some minor organization and naming changes. Also, please add test cases.

Sorry for the long delay in looking at this. I’ve been rather busy at work lately.

**Comment by gagarwal** [comment]
Fixed your code review comments. Still need to add Unit Tests.

**Comment by Dfowj** [comment]
@gagarwal Any update on the unit tests?

**Comment by gagarwal** [comment]
@Dfowj Sorry I didn't got any time to work on this.

**Comment by dfowjtm** [comment]
@gagarwal Checking in on this PR - do you anticipate having the time to write tests for this anytime soon?

**Comment by gagarwal** [comment]
I am sorry, I do not have bandwidth to write them. So sorry.

**Comment by dfowjtm** [comment]
@gagarwal No worries - unfortunately we won't be able to merge this until we have test coverage. However, since getting this merged isn't super urgent, we can wait until you or someone else has time to write the tests for this.

**Pull request closed without merging**